wolfcrypt/src/asn.c

```diff
@@ -2131,20 +2143,31 @@ int GetLength_ex(const byte* input, word32* inOutIdx, int* len, word32 maxIdx,
             minLen = 1 << ((bytes - 1) * 8);
         }
 
-        /* Check the number of bytes required are available. */
-        if ((idx + bytes) > maxIdx) {
-            WOLFSSL_MSG("GetLength - bad long length");
+        /* Check the number of bytes required are available.  At least the
+         * number of bytes specified for a long form length or at least two
+         * bytes for the end-of-content octets for an indefinite form length.
+         */
+        if (((idx + bytes) > maxIdx) ||
+            ((bytes == 0) && (idx > (maxIdx - 2)))) {
```

Comment:
Don't check bytes == 0 but that the tag is ASN_EOC.

Reply:
This change has been reverted / removed with the latest PR change-set.